controller/controller_server/utils.py

```diff
@@ -201,6 +236,35 @@ def generate_csi_create_snapshot_response(new_snapshot, source_volume_id):
     return res
 
 
+def _get_supported_capabilities(volume_capabilities):
+    if volume_capabilities.HasField(config.VOLUME_CAPABILITIES_FIELD_ACCESS_TYPE_MOUNT):
+        return csi_pb2.VolumeCapability(
+            mount=csi_pb2.VolumeCapability.MountVolume(fs_type=volume_capabilities.mount.fs_type),
+            access_mode=csi_pb2.VolumeCapability.AccessMode(mode=volume_capabilities.access_mode.mode))
+
+    elif volume_capabilities.HasField(config.VOLUME_CAPABILITIES_FIELD_ACCESS_TYPE_BLOCK):
+        return csi_pb2.VolumeCapability(
+            mount=csi_pb2.VolumeCapability.BlockVolume(),
+            access_mode=csi_pb2.VolumeCapability.AccessMode(mode=volume_capabilities.access_mode.mode))
```

Comment:
so where do we verify if the volume supports the requested access mode?

Reply:
how can we?

Reply:
AFAIK we currently only support 1 access mode (globally, i.e. for any volume)

Reply:
but how can we check it? is that info available from the storage?

Reply:
I don't think it is relevant for now, since all of our volumes should allow the single access mode we currently support

Reply:
[access_mode validation](https://github.com/IBM/ibm-block-csi-driver/blob/a9950ee1c66276876951b1aa629725ef34ca7cc3/controller/controller_server/utils.py#L64)

Reply:
I see. I think it should be `SUPPORTED_ACCESS_MODES` (plural) - but it's not mandatory for this PR